build-tools/automation/build.groovy

```diff
@@ -87,59 +131,172 @@ timestamps {
             // This is to assure that all tests start from the same state
             sh "killall -9 qemu-system-x86_64 || true"
             sh "rm -rf \$HOME/.android/avd/XamarinAndroidTestRunner.*"
+            if (fileExists(packagePath)) {
+                sh "rm -rf ${packagePath}"
+            }
+
+            sh "mkdir -p ${packagePath}"
         }
 
-        utils.stageWithTimeout('prepare deps', 30, 'MINUTES', XADir, true) {    // Typically takes less than 2 minutes
+        utils.stageWithTimeout('prepare deps', 30, 'MINUTES', XADir, true) {    // Typically takes less than 2 minutes, but can take longer if any prereqs need to be provisioned
+            if (isCommercial) {
+                sh "make prepare-external-git-dependencies"
+
+                utils.stageWithTimeout('provisionator', 30, 'MINUTES', "${commercialPath}/build-tools/provisionator", true) {
```

Comment:
can you nest `stageWithTimeout()` within `stageWithTimeout()`?

Reply:
yes, stages in general can be nested. Jenkins will continue to display them as a single pipeline.  all it knows is that a series of stages have been invoked in some order.  I found it useful for organizing some of the commercial-related stages within its related "parent" stage